packages/components/src/SideDrawer/SideDrawer.tsx

```diff
@@ -38,20 +43,77 @@ interface SideDrawerProps extends PropsWithChildren {
    * Change the scrolling direction of the drawer. Useful for chat-like interfaces.
    */
   readonly scrollDirection?: "normal" | "reverse";
+
+  /**
+   * Element to anchor the drawer to. When provided, the drawer will be positioned
+   * relative to this element instead of the viewport.
+   */
+  readonly anchorElement?: RefObject<HTMLElement>;
+
+  /**
+   * **Use at your own risk:** Custom class names for specific elements. This should only be used as a
+   * **last resort**. Using this may result in unexpected side effects.
+   * More information in the [Customizing components Guide](https://atlantis.getjobber.com/guides/customizing-components).
+   */
+  readonly UNSAFE_className?: {
+    container?: string;
+  };
+
+  /**
+   * **Use at your own risk:** Custom style for specific elements. This should only be used as a
+   * **last resort**. Using this may result in unexpected side effects.
+   * More information in the [Customizing components Guide](https://atlantis.getjobber.com/guides/customizing-components).
+   */
+  readonly UNSAFE_style?: {
+    container?: CSSProperties;
+  };
 }
 
 const variants: Variants = {
   hidden: { x: "100%" },
   visible: { x: 0, transitionEnd: { x: 0 } },
 };
 
-export function SideDrawer({
-  children,
-  onRequestClose,
-  open,
-  variation = "base",
-  scrollDirection,
-}: SideDrawerProps) {
+const useAnchorPosition = (
+  anchorElement: RefObject<HTMLElement> | undefined,
+  open: boolean,
+) => {
+  const [position, setPosition] = useState<{
+    top: number;
+    left: number;
+    width: number;
+  }>({
+    top: 0,
+    left: 0,
+    width: 0,
+  });
+
+  useEffect(() => {
+    if (anchorElement?.current && open) {
+      const updatePosition = () => {
+        const rect = anchorElement.current?.getBoundingClientRect();
+
+        if (rect) {
+          const width = Math.min(420, rect.right);
```

Comment:
I think the `420` should be defined as a constant since right now it is just a "magic number"